content/usgs-packages/sbtools_Get.Rmd

```diff
@@ -0,0 +1,179 @@
+user_id()
+```
+
+When you're done with your session, you can actively logout using the `session_logout`. No arguments are required. If you do not do this, you will be automatically logged out after a certain amount of time or when you close R.
+
+## Inspect and download items
+
+The first inspection step for ScienceBase items is to determine if the item even exists. To do this, use the function `identifier_exists`. The only required argument is `sb_id` which can be either a character string of the item id or an `sbitem`. It will return a logical to indicate if the item exists or not.
+
+```{r sbtools-identifierexists}
+identifier_exists("4f4e4acae4b07f02db67d22b")
+identifier_exists("thisisnotagoodid")
+```
+
+ScienceBase items can be described by alternative identifiers, e.g. digital object identifiers, IPDS codes, etc. They are defined on ScienceBase with a scheme, type, and key. For examples of identifiers, see the "Additional Information | Identifiers" section of [Differential Heating](https://www.sciencebase.gov/catalog/item/580587a2e4b0824b2d1c1f23). 
+
+You can use the function `item_exists` to check whether or not a scheme-type-key tuple already exists. The function has three required arguments - `scheme`, `type`, and `key`. Note that the table of alternative identifiers on ScienceBase is in a different order than this function accepts. On ScienceBase: type, scheme, key. For `item_exists`: scheme, type, key.
+
+```{r sbtools-itemexists}
+# test a made up tuple
+item_exists(scheme = "made", type = "this", key = "up")
+
+# test a tuple from the SB item "4f4e4acae4b07f02db67d22b"
+item_exists(scheme = "State Inventory", type = "UniqueKey", key = "P1281")
+
+# test the same scheme & type with a made up key
+item_exists(scheme = "State Inventory", type = "UniqueKey", key = "1234")
+```
+
+You can create sbitems from just the ScienceBase id. To do this use `as.sbitem`. *why you would use it*
+
+```{r sbtools-as-sbitem}
+antarctica_sbitem <- as.sbitem("4f4e4b24e4b07f02db6aea14")
+class(antarctica_sbitem)
+antarctica_sbitem
+```
+
+Let's inspect various ScienceBase items. There are functions to look at the parent item, metadata fields, sub-items, and associated files. Each of these functions require the id of the sbitem as the first argument. For all of these examples, we are going to use the same sbitem id, "4f4e4b24e4b07f02db6aea14". 
+
+First, let's inspect the parent item. The function to use is `item_get_parent`, and the item id is the only necessary argument.
+
+```{r sbtools-parent}
+ex_id <- "4f4e479de4b07f02db491e34"
+ex_id_parent <- item_get_parent(ex_id)
+ex_id_parent$title
+```
+
+Now, let's see if this item has any children by using the `item_list_children` function. Notice that this function says "list" and not "get" as the previous one did. Functions with "list" only return a few fields associated with each item. Functions with "get" are pulling down all available information, including files, associated with an item.
```

Comment:
well...hmm...no, i'm not seeing any consistent patterns here...

* item_list_children returns a list of sbitems
* item_list_files returns a data.frame
* item_get_parent returns a single sbitem
* item_get_fields returns a list of info or drops from list to vector
* item_get_wfs returns an sp object, e.g., SpatialPolygonsDataFrame
* item_get returns a single sbitem

you could almost claim that `item_get*` expects one of each of whatever it's looking for, except that `item_get_fields(x, 'identifiers')` can return more than one set of identifiers, so...still nope.